src/System.Runtime/tests/System/Text/StringBuilderTests.cs

```diff
@@ -628,10 +628,21 @@ public static void Append_CharArray_Invalid()
             Assert.Throws<ArgumentNullException>("value", () => builder.Append((char[])null, 1, 1)); // Value is null, startIndex > 0 and count > 0
 
             Assert.Throws<ArgumentOutOfRangeException>("startIndex", () => builder.Append(new char[0], -1, 0)); // Start index < 0
-            Assert.Throws<ArgumentOutOfRangeException>("count", () => builder.Append(new char[0], 0, -1)); // Count < 0
 
-            Assert.Throws<ArgumentOutOfRangeException>("count", () => builder.Append(new char[5], 6, 0)); // Start index + count > value.Length
-            Assert.Throws<ArgumentOutOfRangeException>("count", () => builder.Append(new char[5], 5, 1)); // Start index + count > value.Length
+            if (RuntimeDetection.IsNetFramework)
```

Comment:
Ah.
I dunno, I feel a bit like extracting it like this makes it clear it's more of a system dependency, and less something that could be skipped over.

Could still change it if it's too much of an issue?